### PR TITLE
Atomic launch test

### PR DESCRIPTION
**Pull request by sajanrajdev**
Test to execute all of the actions require to launch and setup the protocol's contracts after the KnightingRounds conclude and before the Funding period begins. The test does the following:
- Simulates the KR (For all assets intended to be allowed)
- Mint initial Citadel based on knighting round assets raised
- Send 60% to knighting round for distribution
- Creates a CTDL/wBTC Curve Pool
- LP with 15% of citadel supply + wBTC amount as per initial price ($21)
- Send 25% remaining to treasury vault
- Sends assets acquired through KR to treasury vault
- Initialize and open funding contracts
- Set initial epoch rates and minting start time for SupplySchedule
- finalize() KRs

Additionally, the test functions were removed from the SupplySchedule.sol contract. The BaseFixture was modified to set the epoch rates externally upon setUp().

NOTE: 
- Some price assumptions were made as well as Funding parameters were chosen arbitrarily for the sake of testing
 
TODO:
- Add a user flow to test regular interactions post-launch. The following function can be used for this purpose:
https://github.com/Citadel-DAO/citadel-contracts/blob/44dc8ab64bd67baee50ad5ac3aaac6b46d67fd9c/src/test/AtomicLaunchTest.t.sol#L524-L533

### CODE REVIEW

**Review by Fatimanaz — COMMENTED** [review]
LGTM!

File: src/test/AtomicLaunchTest.t.sol [hunk 1]
```diff
@@ -0,0 +1,644 @@
...
+        assertEq(weth.balanceOf(treasuryVault), govWethBalance);
+
+        // Use 15% of total CTDL to deploy and seed liquidity pool on Curve
+        address[2] memory coins;
+        coins[0] = address(citadel);
+        coins[1] = address(wbtc);
+
+        // NOTE: Parameters acquired from test deployment:
+        // https://etherscan.io/tx/0x20a9182e7644e216d7a26785223fb2947a3ba70998eac4da98a63ec4652b1821
+        address poolAddress = ICurvePoolFactory(curvePoolFactory).deploy_pool(
```
Comment (dapp-whisperer) [inline src/test/AtomicLaunchTest.t.sol:349]: Are these the default params for volatile pair? @sajanrajdev

**Comment by dapp-whisperer** [comment]
Notes for self:
- removed test functions `_setEpochRates()`, `getMintableDebug()`
- adds all funding assets to test env, but they are not used yet

### KR
- all KR assets are covered, including ETH variant

### Funding
- Funding for cvx, wbtc, badger
- Deploys median oracles for cvx, wbtc, badger
- All oracle prices are set by EOA fiat (good abstraction)

**Comment by dapp-whisperer** [comment]
Very clean and nice abstraction, LGTM

**Review by dapp-whisperer — APPROVED** [review]

File: src/test/AtomicLaunchTest.t.sol (hunk 1 above)
Comment (sajanrajdev) [reply src/test/AtomicLaunchTest.t.sol:349]: That's right, from UI:
image

**Comment by sajanrajdev** [comment]
This PR closes https://github.com/Citadel-DAO/citadel-contracts/issues/40

**Comment by dapp-whisperer** [comment]
Volatile curve pool parameters for records
image

> Only change made from the test deployment was the initial price which was set as: Current wBTC price/Current Citadel price - aka ~$30k/$21
> There seems to be a bug in their UI because it says current external rate for TOKEN_1/TOKEN_2, when it should be the other way around
> Ayush confirmed and only doing it the other way around worked

**Pull request merged**